File: src/preprocess.R
# src/preprocess.R
library(raster)
library(ggplot2)

calculate_ndvi <- function(input_tif, output_png) {
  raster_data <- stack(input_tif)
  ndvi <- (raster_data[[4]] - raster_data[[1]]) / (raster_data[[4]] + raster_data[[1]])
  
  ggplot() +
    geom_raster(data = as.data.frame(rasterToPoints(ndvi)), aes(x = x, y = y, fill = layer)) +
    scale_fill_gradient(low = "yellow", high = "green") +
    ggtitle("NDVI Map")
  ggsave(output_png)
}

calculate_ndvi("data/landcover.tif", "outputs/ndvi_map.png")

File: src/preprocess.py
# src/preprocess.py
import rasterio
import numpy as np
import matplotlib.pyplot as plt

def calculate_ndvi(input_tif, output_png):
    with rasterio.open(input_tif) as src:
        nir = src.read(4)
        red = src.read(1)
    
    ndvi = (nir.astype(float) - red.astype(float)) / (nir + red)
    
    plt.imshow(ndvi, cmap='RdYlGn')
    plt.colorbar(label='NDVI')
    plt.title("NDVI Map")
    plt.savefig(output_png)
    plt.show()

calculate_ndvi("data/landcover.tif", "outputs/ndvi_map.png")